network/peers/subnets.go

```diff
@@ -29,10 +29,12 @@ func (si *subnetsIndex) UpdatePeerSubnets(id peer.ID, s records.Subnets) bool {
 	defer si.lock.Unlock()
 
 	existing, ok := si.peerSubnets[id]
+	var diff map[int]byte
 	if !ok {
-		existing = make([]byte, 0)
+		diff = s.ToMap()
+	} else {
+		diff = existing.DiffSubnets(s)
 	}
-	diff := records.DiffSubnets(existing, s)
```

Comment:
For better readability, this can be simplified to something like this:
```go
diff := s.ToMap()
if ok {
    diff = existing.DiffSubnets(s)
}
```
unless we are doing it **a lot** (which would create garbage for gc).

Reply:
Right, that's just some styling preference I guess ;)